docs/units/distributions.rst

```diff
@@ -0,0 +1,243 @@
+.. |quantity| replace:: :class:`~astropy.units.Quantity`
+.. |distribution| replace:: :class:`~astropy.units.Distribution`
+
+.. _unit_distributions:
+
+Distributions
+*************
+
+In addition to |quantity|, astropy provides a |distribution| object to represent
+statistical distributions in a form that acts as a drop-in replacement for
+|quantity| or regulay Numpy arrays. USed in this manner, |distribution| provides
+uncertainty propogation at the cost of additional computation.  It can also
+more generally represent sampled distributions for e.g., Monte Carlo calculation
+techniques.
+
+The core object for this feature is the |distribution|.  Currently, all
+such distributions are Monte Carlo sampled.  While this means each distribution
+may take more memory, it allows arbitrarily complex operations to be performed
+on distributions while maintaining their correlation structure. Some specific
+well-behaved distributions (e.g., `~astropy.units.NormalDistribution`) have
+analytic forms which may eventually enable a more compact/efficient
+representation, but this is not currently implemented.

```

Comment:
Just a note about this - if we ever do this we'll need to make sure we still deal with correlated samples.

Reply:
My `Variable` class does do that. I really should try to resurrect it; it is at a level at least equal to `Distribution`, with the only uncertainy about how far to continue to propagate dependencies.

Reply:
@mhvk - I think the thought was to do exactly that once we've had some API experience with this. I guess this may provide a place to do that! (I still don't like the name `Variable` though :wink:)

Reply:
Indeed. And the name of course is open for discussion!